src/main/java/com/awesomeorg/airlineservice/service/TicketService.java

```diff
@@ -1,12 +1,81 @@
 package com.awesomeorg.airlineservice.service;
 
+import com.awesomeorg.airlineservice.entity.Ticket;
+import com.awesomeorg.airlineservice.exceptions.TicketAlreadyExistsException;
+import com.awesomeorg.airlineservice.exceptions.TicketNotFoundException;
+import com.awesomeorg.airlineservice.protocol.TicketQuery;
+import com.awesomeorg.airlineservice.protocol.UpdateTicketRequest;
 import com.awesomeorg.airlineservice.repository.TicketRepository;
 import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.Optional;
+
 @Service
 @RequiredArgsConstructor
 public class TicketService {
 
     private final TicketRepository ticketRepository;
-}
+
+    public Optional<Ticket> findTicketById(final Long ticketId) {
+        if (ticketId == null) {
+            return Optional.empty();
+        }
+        return ticketRepository.findById(ticketId);
+    }
+
+    public Page<Ticket> findFreeTicket(TicketQuery query, PageRequest pageRequest) {
+        LocalDate dateOfFlight = query.getDateOfFlight();
+        return ticketRepository.findFreeTicket(dateOfFlight, pageRequest);
+    }
+
+
+
+    public Ticket createTicket(TicketQuery request) {
+        LocalDate dateOfPurchase = request.getDateOfPurchase();
+
+        // Check if ticket already exists
+        final Page<Ticket> existingTickets = ticketRepository.findFreeTicket(dateOfPurchase, PageRequest.of(0, 25));
```

Comment:
Still, it doesn't make much sense.
If one customer buys ticket for that date, then the other customer wouldn't be able to buy for the same date. 
There is some flaw in the overall structure. Reservation requires a ticket, however, they almost have the same fields, except seat, price and date of purchase. I would actually combine them, so it's easier